File: in_Python/1089 Duplicate Zeros.py
class Solution(object):
    def duplicateZeros(self, arr):
        """
        :type arr: List[int]
        :rtype: None Do not return anything, modify arr in-place instead.
        """
        dict = {}
        num_zero = 0
        for i in range(len(arr)):
            if arr[i] == 0:
                num_zero += 1
            else:
                dict[i + num_zero] = arr[i]
        
        for idx in range(len(arr)):
            if idx in dict:
                arr[idx] = dict[idx]
            else:
                arr[idx] = 0
        
        

File: in_Java/1089 Duplicate Zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        int czeros = 0;
        for(int i : arr){
            if(i == 0) czeros++;
        }
        int len = arr.length + czeros;
        for(int i = arr.length -1, j = len-1; i < j; i--,j--){
            if(arr[i] != 0){
                if(j < arr.length) arr[j] = arr[i];
            }else{
                if(j < arr.length) arr[j] = arr[i];
                j--;
                if(j < arr.length) arr[j] = arr[i];
            }
        }
    }
}